drivers/serial/uart_ns16550.c

```diff
@@ -505,8 +505,21 @@ static int uart_ns16550_poll_in(struct device *dev, unsigned char *c)
 static void uart_ns16550_poll_out(struct device *dev,
 					   unsigned char c)
 {
+	struct uart_ns16550_dev_data_t *data = DEV_DATA(dev);
+
 	/* wait for transmitter to ready to accept a character */
-	while ((INBYTE(LSR(dev)) & LSR_THRE) == 0) {
+	while (1) {
```

Comment:
I think you need to rebase. When you do, please note the changes in commit 1cd5578 `serial: ns16550: Simplify poll out` . The `while (1)` was removed in favor of the line 509 you see here. Will you need to bring it back?

Reply:
Rebased. Seems reverting previous commit was the most sensible way to do this change and keep the code readable.